### PR TITLE
test(warehouse): pump more test for warehouse module

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

Some code cleanup.

Added a new set of test cases for the warehouse module around:
- Warehouse schemas
- Warehouse GRPC modules
- Warehouse archiving
- Warehouse table uploads
- Warehouse uploads test

## Notion Ticket

https://www.notion.so/rudderstacks/Warehouse-tests-97aa9f5fba024556b1fe2e6f3d604562

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: warehouse/table_upload_test.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/ory/dockertest/v3"
+	"github.com/rudderlabs/rudder-server/testhelper"
+	"github.com/rudderlabs/rudder-server/testhelper/destination"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+func tableUploadsTestSQLStatement() string {
```
Comment (achettyiitr) [inline warehouse/table_upload_test.go:18]: Move this to testdata files.

File: warehouse/warehousegrpc.go [hunk 2]
```diff
@@ -11,11 +11,11 @@ import (
 	proto "github.com/rudderlabs/rudder-server/proto/warehouse"
 )
 
-type warehousegrpc struct {
+type warehouseGrpc struct {
```
Comment (achettyiitr) [inline warehouse/warehousegrpc.go:14]: warehouseGRPC

Comment (achettyiitr) [reply warehouse/warehousegrpc.go:14]: Done.

File: warehouse/table_upload_test.go (hunk 1 above)
Comment (achettyiitr) [reply warehouse/table_upload_test.go:18]: Done.

**Review by lvrach — APPROVED** [review]
Approve, with some notes to revisit later:

* Directly populating the database using SQL statements as a precondition for a test
* Usage of config or DI in place of `.Setenv`, as environment variables can cause harm when a test is run concurrently.

**Pull request merged**